c7n/policy.py

```diff
@@ -319,6 +319,9 @@ def run(self, *args, **kw):
             ctx.metrics.put_metric(
                 "ResourceCount", len(resources), "Count", Scope="Policy"
             )
+            if len(resources) and COST_ANNOTATION_KEY in resources[0]:
```

Comment:
this I'm a little less clear about, it feels quite odd having the policy trying to have awareness of individual filters and poking at things, we could potentially have the filter write it, but that leaves other oddities wrt to multiple filters. I think for now its best left out.

Reply:
I am struggling with whether follow your suggestion or not.
- in the use case from my org, it does not have too much value as a filter, rather, what we really want is the annotation, so that we may have the info in the notification and the dashboard (via Grafana).
- on the other hand, the implementation has some bad smell, yet I don't know how to make it better. I am not a fan of the idea that having the filter write the metrics.

So, I am going to keep it this way with a TODO note to remind someone smarter to refactor it in the future.